daemon/images/image_push.go

```diff
@@ -10,11 +11,16 @@ import (
 	"github.com/docker/docker/api/types/registry"
 	"github.com/docker/docker/distribution"
 	progressutils "github.com/docker/docker/distribution/utils"
+	"github.com/docker/docker/errdefs"
 	"github.com/docker/docker/pkg/progress"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
 )
 
 // PushImage initiates a push operation on the repository named localName.
-func (i *ImageService) PushImage(ctx context.Context, ref reference.Named, metaHeaders map[string][]string, authConfig *registry.AuthConfig, outStream io.Writer) error {
+func (i *ImageService) PushImage(ctx context.Context, ref reference.Named, platform *ocispec.Platform, metaHeaders map[string][]string, authConfig *registry.AuthConfig, outStream io.Writer) error {
+	if platform != nil {
+		return errdefs.NotImplemented(errors.New("selecting platform is not supported with graphdriver backed image store"))
+	}
```

Comment:
Do we have the platform information of the image available here? Could we make it only fail if the given platform doesn't match the image? That would make it "work" with the graphdriver.


The error should probably be a `InvalidParameter` otherwise we may be returning a `501`, which is not currently documented in the API, so would (technically?) mean a change in the API 🙈 
<img width="196" alt="Screenshot 2024-06-07 at 15 51 01" src="https://github.com/moby/moby/assets/1804568/03575adb-724a-49b5-b289-f7886e90c2e4">

Reply:
Hmm good idea. Changed it so it works if the platform matches, and returns an invalid parameter otherwise.